src/poetry/utils/cache.py

```diff
@@ -0,0 +1,250 @@
+from __future__ import annotations
```

Comment:
Let's move this module into a top-level package `poetry.cache`. Ideally with relevant bits split out into its own modules.